components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -960,7 +959,16 @@ private static void setConsentRequiredScopesToOAuthParams(AuthenticatedUser user
 
         try {
             String consentRequiredScopes = StringUtils.EMPTY;
-            List<String> allowedOAuthScopes = getAllowedOAuthScopes(params);
+            Set<String> allowedScopes = params.getScopes();
+            List<String> allowedOAuthScopes = new ArrayList<>();
+            String[] oidcScopes = oAuthAdminService.getScopeNames();
+            //filter out OIDC scopes
+            List<String> oidcScopeList = new ArrayList<>(Arrays.asList(oidcScopes));
+            for (String scope : allowedScopes) {
+                if (!oidcScopeList.contains(scope)) {
+                    allowedOAuthScopes.add(scope);
+                }
+            }
```

Comment:
Would be good extract this to a private method that does indicates we are removing OIDC scopes from consent required scope list

Reply:
added here https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1950/files#diff-b6225573273ddb543f5bf377f14d808d5c96f4d4a0aa99359f843b2bda55fd36R917